kolibri/core/assets/src/composables/useScrollPosition.js

```diff
@@ -0,0 +1,77 @@
+import { computed } from 'kolibri.lib.vueCompositionApi';
+import { get } from '@vueuse/core';
+
+export default function useScrollPosition() {
+  let _scrollPositions = {};
+
+  const scrollPosition = computed({
+    get() {
+      // Use key set by Vue Router on the history state.
+      const key = (window.history.state || {}).key;
+      const defaultPos = { x: 0, y: 0 };
+      if (key && _scrollPositions[key]) {
+        return _scrollPositions[key];
+      }
+      return defaultPos;
+    },
+    set(newLocation) {
+      // reference location can be { x: 0, y: 0 } coordinates using a Object,
+      // or a string i.e. a css selector
+      // for any instance
+      let x, y;
+      if (newLocation instanceof String) {
+        if (document.querySelector(newLocation)) {
+          const position = document.querySelector(newLocation).getBoundingClientRect();
+          x = position.left;
+          y = position.top;
+        } else {
+          x = 0;
+          y = 0;
+        }
+      } else if (newLocation instanceof Object) {
+        if (!isNaN(newLocation.x)) {
+          x = newLocation.x;
+        } else {
+          x = 0;
+        }
+        if (!isNaN(newLocation.y)) {
+          y = newLocation.y;
+        } else {
+          y = 0;
+        }
+      } else {
+        x = 0;
+        y = 0;
+      }
+      const key = (window.history.state || {}).key;
+      // Only set if we have a vue router key on the state,
+      // otherwise we don't do anything.
+      if (key) {
+        _scrollPositions[window.history.state.key] = { x: x, y: y };
```

Comment:
Looks like `window.history.state.key` could just be `key` in this line (not a blocker)